### PR TITLE
CubicalPersistence: handle 'vertices' input, special case 1d

### PR DESCRIPTION
**Pull request by mglisse**
(this comes on top of #784)
For now I am not giving persistence_on_a_line a specific non-sklearn interface, that can come later.

### CODE REVIEW

File: src/python/gudhi/_persline.cc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    Gudhi::persistent_cohomology::compute_persistence_of_function_on_line(r, [&](T b, T d){ dgm.push_back({b, d}); });
+  }
+  return py::array(py::cast(std::move(dgm)));
+}
+
+PYBIND11_MODULE(_persline, m) {
+  py::bind_vector<Vf>(m, "VectorPairFloat" , py::buffer_protocol());
+  py::bind_vector<Vd>(m, "VectorPairDouble", py::buffer_protocol());
+  m.def("persistence_on_a_line", fun<float>, py::arg().noconvert());
+  m.def("persistence_on_a_line", fun<double>);
```
Comment (VincentRouvreau) [inline src/python/gudhi/_persline.cc:52]: This function is hidden, I am ok with it. Just a reminder to not forget the documentation. For the moment, we have:
```python
from gudhi._persline import persistence_on_a_line
help(persistence_on_a_line)

# Help on built-in function persistence_on_a_line in module gudhi._persline:
# 
# persistence_on_a_line(...) method of builtins.PyCapsule instance
#     persistence_on_a_line(*args, **kwargs)
#     Overloaded function.
#     
#     1. persistence_on_a_line(arg0: numpy.ndarray[numpy.float32]) -> numpy.ndarray
#     
#     2. persistence_on_a_line(arg0: numpy.ndarray[numpy.float64]) -> numpy.ndarray
```

File: src/python/test/test_sklearn_cubical_persistence.py [hunk 2]
```diff
@@ -39,21 +29,38 @@ def test_simple_constructor_from_top_cells_list():
...
+    assert len(diags) == 10
+    np.testing.assert_array_equal(diags[2], [[8, 13], [0, 15], [0, np.inf]])
+
+
+def test_1d():
+    a = np.array([2, 4, 3, 5])
+    r = np.array([[3, 4], [2, np.inf]])
+    np.testing.assert_array_equal(r, CubicalPersistence(0).fit_transform([a])[0])
+    np.testing.assert_array_equal(r, CubicalPersistence(0).fit_transform([a.astype("float32")[::-1]])[0])
+    np.testing.assert_array_equal(r, CubicalPersistence([0]).fit_transform([a.astype("float64")])[0][0])
```
[the remark this answers is not in the document]
Comment (mglisse) [reply src/python/test/test_sklearn_cubical_persistence.py]: done

File: src/python/gudhi/_persline.cc (hunk 1 above)
Comment (mglisse) [reply src/python/gudhi/_persline.cc:52]: I am not sure what you mean exactly. There is no way a user is supposed to know that this function even exists, the user documentation is in ComplexPersistence. Do you mean you want a comment in the file for us to remember what the function does? Or do you mean that if/when we make this function public, we will need to document it?

Comment (VincentRouvreau) [reply src/python/gudhi/_persline.cc:52]: In the title of the PR, you say
 > For now I am not giving persistence_on_a_line a specific non-sklearn interface, that can come later.
When the time will come, we shall not forget to document the function `persistence_on_a_line`

**Pull request merged**